vllm/block.py

```diff
@@ -59,6 +59,10 @@ def __len__(self) -> int:
     def __getitem__(self, key):
         return self._blocks[key]
 
+    def __iter__(self) -> Iterator[PhysicalTokenBlock]:
+        for i in range(len(self)):
+            yield self[i]
```

Comment:
I think replacing the internally generated method (presumably in C) with an explicit implementation might harm the performance. How about defining it like this?
```suggestion
    if TYPE_CHECKING:
        def __iter__(self) -> Iterator[PhysicalTokenBlock]:
            raise RuntimeError("Method should be automatically generated")
```